subsys/net/l2/ieee802154/ieee802154_frame.c

```diff
@@ -330,7 +330,7 @@ validate_mac_command(struct ieee802154_mpdu *mpdu, uint8_t *buf, uint8_t *length
 		break;
 	case IEEE802154_CFI_ORPHAN_NOTIFICATION:
 		comp = 1U;
-		src = IEEE802154_EXT_ADDR_LENGTH;
+		src = IEEE802154_ADDR_MODE_EXTENDED;
```

Comment:
What about line  345 ?

Reply:
Oh right, should be changed as well. I'll do it.

While looking again at this file, the logic really seems broken because sometimes we do:
`dst = IEEE802154_ADDR_MODE_SHORT | IEEE802154_ADDR_MODE_EXTENDED;`
Which is the equivalent currently of:
`dst = 0x2 | 0x3;`
It won't work with short addressing. I'll fix it in another patch.